### ISSUE TITLE
Calling reset() on an object is deprecated in (Statement.php on line 690) PHP 8.1.4

### ISSUE THREAD

**Issue by nazaridotco** [yajra/pdo-via-oci8#108]
### Summary of problem or feature request

I'm using Laravel OCI8 v9 which I have many deprecation logs for Calling `reset()` function for PHP 8.1.4: 
```
local.WARNING: reset(): Calling reset() on an object is deprecated in *\vendor\yajra\laravel-pdo-via-oci8\src\Pdo\Oci8\Statement.php on line 690
```
I already post comment in Laravel OCI8 [here ](https://github.com/yajra/laravel-oci8/issues/692#issuecomment-1152888449)

### Code snippet of problem

The following code fixed my issue hope do the same for other, line 690, 691of `yajra\laravel-pdo-via-oci8\src\Pdo\Oci8\Statement.php` need to be update from this: 
```
 if ((is_array($row) || is_object($row)) && is_resource( reset($row) )) {
                $stmt = new self(reset($row)), $this->connection, $this->options);
``` 
To this: 
```
 if ((is_array($row) || is_object($row)) && is_resource( current( ( (array) $row) ) )) {
                $stmt = new self(current(((array) $row)), $this->connection, $this->options);
```
The above code fixed my deprecation logs. this is my `Log::alert(reset($row) . ' ' . current(((array) $row)) );`
`[2022-06-11 09:12:09] local.ALERT: BACKUP BACKUP`

### System details
- Operating System: Windows 10 pro
- PHP Version: 8.1.4
- PDO-VIA-OCI8 Version: 3.2.2
- Laravel version: 9.17

**Comment by yajra** [comment]
Would you be able to review this PR https://github.com/yajra/pdo-via-oci8/pull/104 if it fixes the issue? Will merge when you confirm it's working.

**Comment by nazaridotco** [comment]
Yes its working according to my test in this [comment ](https://github.com/yajra/laravel-oci8/issues/692#issuecomment-1152888449)

**Comment by yajra** [comment]
Released on v3.2.3 thanks!
